src/core/request.js

```diff
@@ -59,31 +59,31 @@ Request.prototype.request = function(servers, callback_) {
   this.on('error', function() {});
   this.emit('request', this.remote);
 
-  function doRequest() {
-    if (Array.isArray(servers)) {
-      servers.forEach(function(server) {
-        self.setServer(server);
-        self.remote.request(self);
-      }, self);
-    } else {
-      self.remote.request(self);
+  const timeout = setTimeout(() => {
+    if (typeof callback === 'function') {
+      callback(new RippleError('tejTimeout'));
```

Comment:
Do you need a `return` here? Otherwise it looks like it will emit `timeout` and if no listener is registered it will raise an uncaught exception right?